### ISSUE TITLE
`Mime::Type::InvalidMimeType` errors 500 (should 406) for API-only applications

### ISSUE THREAD

**Issue by lorennorman** [rails/rails#38977]
TL;DR: this line: https://github.com/rails/rails/blob/157920aead96865e3135f496c09ace607d5620dc/actionpack/lib/action_dispatch/middleware/debug_exceptions.rb#L68

...should be either...
```ruby
return render_for_api_request(Mime[:text], wrapper)
```
...or...
```ruby
content_type = Mime[:text]
```

### Steps to reproduce
**Option 1:**
1. Clone this repo: https://github.com/lorennorman/rescue-middleware-test
2. `rails db:create db:migrate`
3. `rails t`

**Option 2:**
1. Generate a Rails 6.0.2.2 app
2. rails db:create db:migrate
3. Add a route: `root to: "basic#index"`
4. Add `index` action to ApplicationController:
```ruby
  def index
    render text: "hello"
  end
```
5. Enable `show_exceptions` and `api_only` in `config/environments/test.rb`:
```ruby
  config.action_dispatch.show_exceptions = true
  config.api_only = true
```

6. Add this test `test/integration/invalid_mime_type_middleware_test.rb`:
```ruby
require 'test_helper'

class InvalidMimeTypeMiddlewareTest < ActionDispatch::IntegrationTest
  test 'invalid mime type should 406' do
    get '/', headers: { accept: "../../../etc/passwd" }
    assert_response :not_acceptable
  end
end
```

7. Run it: `rails t`

### Expected behavior
tl;dr Rails should respond with a 406, passing the simple test.

Full description:
1. `ActionDispatch::ShowExceptions` should rescue `Mime::Type::InvalidMimeType`...
2. ...look it up in `ActionDispatch::ExceptionWrapper.rescue_responses`...
3. ...and set the response to 406, `NOT_ACCEPTABLE`

### Actual behavior
tl;dr Rails responds with 500, failing the test.

Full description:
1. These lines contain the bug: https://github.com/rails/rails/blob/157920aead96865e3135f496c09ace607d5620dc/actionpack/lib/action_dispatch/middleware/debug_exceptions.rb#L64-L76

Line 68 should return, or simply set the `content_type` variable. Doing neither causes Line 71 to call `api_request?(nil)`, which in turn makes...

2. These lines explode: https://github.com/rails/rails/blob/157920aead96865e3135f496c09ace607d5620dc/actionpack/lib/action_dispatch/middleware/debug_exceptions.rb#L181-L183

...with a `NoMethodError` on `.html?` for `NilClass`.

3. All of this gets hidden because it is already sensitive exception handling code, so the real error gets swallowed, the end user just gets a 500, and the developer is left without a stack trace to the real error. Subtle!

### System configuration
**Rails version**: 6.0.2.2

**Ruby version**: 2.6.3p62
